eng/testing/xunit/xunit.console.targets

```
@@ -87,11 +87,12 @@
     </ItemGroup>
 
     <!-- Workaround for https://github.com/xunit/xunit/issues/1651 -->
-    <ItemGroup Condition="'$(ArchiveTests)' != 'true'">
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\net452\xunit.runner.utility.net452.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\net452\xunit.runner.reporters.net452.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\netcoreapp2.1\xunit.runner.utility.netcoreapp10.dll" />
-      <None Remove="$(Pkgxunit_runner_visualstudio)\build\netcoreapp2.1\xunit.runner.reporters.netcoreapp10.dll" />
+    <ItemGroup Condition="'$(Pkgxunit_runner_visualstudio)' != '' AND '$(TargetFrameworkIdentifier)' == '.NETFramework'">
+      <!-- Remove `xunit.abstractions.dll` that was added to @(None) by xunit.runner.visualstudio -->
+      <_xunitAbstractionsRemove Include="@(None->WithMetadataValue('Filename', 'xunit.abstractions'))" />
+      <None Remove="@(_xunitAbstractionsRemove)" />
     </ItemGroup>
+    <Error Condition="'$(Pkgxunit_runner_visualstudio)' != '' AND '$(TargetFrameworkIdentifier)' == '.NETFramework' AND '@(_xunitAbstractionsRemove)' == ''"
+           Text="Remove workaround above and GeneratePathProperty from the xunit.runner.visualstudio PackageReference as they are no longer needed"/>
```

I see that you inverted the ArchiveTests condition check: We only add the runner DLLs when ArchiveTests is true, instead of removing them when ArchiveTests is false. I can see this is a slight improvement especially since there's no need to check for hardcoded versions in the files.

And I can see that this error you're adding will only trigger when we finally get the issue fixed and xunit.abstractions is no longer getting manually removed via line 92. Am I reading it correctly?

Finally, I assume we are _still_ considering this a workaround for https://github.com/xunit/xunit/issues/1651 even though that issue has just been fixed, because we can't simplyu update runtime to consume that latest xunit version that contains that fix. Is this assumption correct?

TIL you can get the path from a package reference using the GeneratePathProperty attribute: https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#generatepathproperty

Btw - super tiny and annoying nit:

```suggestion
           Text="Remove workaround above and GeneratePathProperty from the xunit.runner.visualstudio PackageReference as they are no longer needed." />
```

---

> And I can see that this error you're adding will only trigger when we finally get the issue fixed and xunit.abstractions is no longer getting manually removed via line 92. Am I reading it correctly?

Yep

> Finally, I assume we are still considering this a workaround for https://github.com/xunit/xunit/issues/1651 even though that issue has just been fixed, because we can't simplyu update runtime to consume that latest xunit version that contains that fix. Is this assumption correct?

Yeah, that's what I explained here: https://github.com/dotnet/runtime/pull/96826#discussion_r1458021352